### ISSUE TITLE
Replacing the normalizer in unit tests (with parse5 perhaps)

### ISSUE THREAD

**Issue by mightyiam** [wymeditor/wymeditor#497]
[Our current `normalizeHtml`](https://github.com/wymeditor/wymeditor/blob/89165131f2bee29f64dc3f3de5c508b0b5db1f5f/src/test/unit/utils.js#L51) has a design flaw which compromises the reliability of most of the unit testing framework.

Most of our unit tests use [`htmlEquals`](https://github.com/wymeditor/wymeditor/blob/89165131f2bee29f64dc3f3de5c508b0b5db1f5f/src/test/unit/utils.js#L174). `htmlEquals` is passed an argument, `expected`—a string from the unit test—and should compare it with HTML that it fetches from the WYMeditor.

Here is a simplification of the steps `htmlEquals` perfoms:
1. The output HTML of the WYMeditor is fetched (line 181). This is a string of HTML which was already parsed by the parser. So we get a string of parsed HTML.
2. It then uses jQuery to get a jQuery object of DOM nodes constructed from that string of HTML (line 190).
3. It then passes each of the nodes in the jQuery object through `normalizeHtml`, in order to get a string of normalized HTML. The normalization process is supposed to provide a string of HTML that is identical in essence. The differentiation between essence and form is, for example, that one browser can provide `<OL id="foo" class="bar">` and the other can provide `<ol class="bar" id="foo">`. These are different in form and identical in essence. The normalizer should always provide an expected result (e.g. lowercase tags and sorted attributes).

So, in short:
1. Fetch string of HTML from WYMeditor.
2. Make DOM nodes out of it.
3. Get normalized string of HTML out of those nodes.

The design flaw is that the string of HTML gets turned into DOM nodes. This should not happen. This invalidates the integrity of the tests.

Instead, what should happen is this:
1. Fetch string of HTML from the WYMeditor.
2. Get a normalized string of HTML from that string of HTML, using a normalizer that processes a string of HTML.
